hbase_storage/src/main/java/com/splicemachine/si/impl/server/SICompactionStateMutate.java

```diff
@@ -0,0 +1,125 @@
+package com.splicemachine.si.impl.server;
+
+import com.splicemachine.hbase.CellUtils;
+import com.splicemachine.primitives.Bytes;
+import com.splicemachine.si.api.txn.Txn;
+import com.splicemachine.si.api.txn.TxnView;
+import com.splicemachine.si.constants.SIConstants;
+import com.splicemachine.storage.CellType;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.KeyValue;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+
+class SICompactionStateMutate {
+    private SortedSet<Cell> dataToReturn;
+    private final boolean purgeDeletedRows;
+    private long maxTombstoneTimestamp;
+
+    SICompactionStateMutate(boolean purgeDeletedRows) {
+        this.purgeDeletedRows = purgeDeletedRows;
+        this.dataToReturn = new TreeSet<>(KeyValue.COMPARATOR);
+        this.maxTombstoneTimestamp = 0;
+    }
+
+    private boolean isSorted(List<Cell> list) {
```

Comment:
You could use Guava here: `Ordering.from(KeyValue.COMPARATOR).isOrdered(list);`

Reply:
I wanted to, but I could not make it work without adding a new dependency and was wondering if it was worth it.